src/forms/venueV2Schema.ts

```diff
@@ -0,0 +1,57 @@
+import firebase from "firebase/app";
+import * as Yup from "yup";
+
+import { createSlug } from "api/admin";
+
+import { mustBeMinimum } from "forms/common";
+import { commonTitleSchema } from "forms/commonTitleSchema";
+
+export interface VenueV2SchemaShape {
+  name: string;
+  subtitle?: string;
+  description?: string;
+
+  bannerImageUrl?: string;
+  logoImageUrl: string;
+}
+
+export const venueV2Schema = Yup.object()
+  .shape<VenueV2SchemaShape>({
+    name: commonTitleSchema.when(
+      "$editing",
+      (editing: boolean, schema: Yup.StringSchema) =>
+        !editing
+          ? schema
+              .test(
+                "name",
+                "Must have alphanumeric characters",
+                (val: string) => createSlug(val).length > 0
+              )
+              .test(
+                "name",
+                "This venue name is already taken",
+                async (val: string) =>
+                  !val ||
+                  !(
+                    await firebase
```

Comment:
we might add smth like `// @debt Replace with a function from api/venues` as done in `src/forms/worldStartSchema.ts`

Reply:
Even better, add it to
- https://github.com/sparkletown/internal-sparkle-issues/issues/1462

Reply:
Even better, add it to
- https://github.com/sparkletown/internal-sparkle-issues/issues/1462

Reply:
btw, is smth = shaking my head variation?

Reply:
nah, i'm using it as `something` 
![image](https://user-images.githubusercontent.com/56254806/142609379-2d228813-a6d5-4642-b029-a34de24aa933.png)

Reply:
`something` reads better than mistakenly getting `smth` as `shaking my thick head` or worse 🤪